drivers/hmis_external_apis/lib/tasks/import.rake

```diff
@@ -115,4 +115,220 @@ namespace :import do
+      entry_date_time = row.field_value('OPEN_DT')
+      # Skip if any required fields are missing
+      unless case_id && referral_id && entry_date_time
+        skipped_case_ids << case_id
+        next
+      end
+      # Skip if Referral ID already processed (shouldn't happen)
+      if referral_id_to_case_id.key?(referral_id)
+        Rails.logger.info "Referral ID #{referral_id} already seen. Skipping."
+        next
+      end
+
+      referral_id_to_case_id[referral_id] = case_id
+      referral_id_to_entry_date[referral_id] = entry_date_time
+    end
+
+    # Read Household file. Each row is a household member. They are linked together by HOUSEHOLD_ID.
+    # Later on, we will create an Enrollment for each household member. For now, just store the mappings. The ReferralID is used to cross reference with the main Export file.
+    hh_export_file = HmisExternalApis::TcHmis::Importers::Loaders::XlsxFile.new(filename: "#{dir}/HCM-EXPORT-HH.xlsx", sheet_number: 0, header_row_number: 1)
+    referral_id_to_hh_id = {} # just to check for duplicates
+    referral_id_to_household = {}
+    mci_ids = Set.new
+    hh_export_file.each do |row|
+      referral_id = row.field_value('ReferralID')
+      household_id = row.field_value('HOUSEHOLD_ID')
+      mci_id = row.field_value('MCI_ID')
+      mci_ids << mci_id
+      relation_to_id = row.field_value('RELATION_TO_ID')
+
+      unless household_id && referral_id && mci_id
+        skipped_case_ids << case_id
+        next
+      end
+
+      # Raise if Referral ID has multiple households, shouldn't happen
+      raise "Referral ID #{referral_id} linked to multiple households. unexpected." if referral_id_to_hh_id.key?(referral_id) && referral_id_to_hh_id[referral_id] != household_id
+
+      referral_id_to_hh_id[referral_id] = household_id
+
+      referral_id_to_household[referral_id] ||= []
+      referral_id_to_household[referral_id] << { mci_id: mci_id, relation_to_id: relation_to_id }
+    end
+
+    # Find Clients. No special care taken to situation where there are multiple clients with the same MCI ID, just pick one.
+    mci_id_to_client = HmisExternalApis::ExternalId.mci_ids.where(value: mci_ids).preload(:source).index_by(&:value)
+
+    # Build Enrollments
+    enrollments = []
+    # Map { Case ID => Enrollment }, so we can match to Case Notes file later. Maps to HoH or just some member if there is no HoH.
+    case_id_to_hoh_enrollment = {}
+    # MCI IDs that were skipped because we couldn't find them in our system.
+    skipped_mci_ids = []
+
+    referral_id_to_entry_date.each do |referral_id, entry_date_time|
+      case_id = referral_id_to_case_id[referral_id]
+
+      household = referral_id_to_household[referral_id]
+      if !household
+        Rails.logger.info "Household not found for referral ID #{referral_id}. Skipping."
+        skipped_case_ids << case_id
+        next
+      end
+
+      household_id = Hmis::Hud::Base.generate_uuid
+      household.each do |member|
```

Comment:
should we check for duplicate household members? (one client in the same household multiple times)

Reply:
I dont think so, that's an acceptable data quality issue. Similarly I am not checking for multiple HoHs or no HoHs. Those are things they can clean up, it is a small enough data set.